src/libLookup/Lookup.cpp

```diff
@@ -1597,73 +1597,97 @@ bool Lookup::ProcessSetTxBlockFromSeed(const vector<unsigned char>& message,
-                "txBlock.GetHeader().GetGasUsed(): "
-                    << txBlock.GetHeader().GetGasUsed());
-      LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                "txBlock.GetHeader().GetBlockNum(): "
-                    << txBlock.GetHeader().GetBlockNum());
-      LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                "txBlock.GetHeader().GetNumMicroBlockHashes(): "
-                    << txBlock.GetHeader().GetNumMicroBlockHashes());
-      LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                "txBlock.GetHeader().GetNumTxs(): "
-                    << txBlock.GetHeader().GetNumTxs());
-      LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                "txBlock.GetHeader().GetMinerPubKey(): "
-                    << txBlock.GetHeader().GetMinerPubKey());
-      LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                "txBlock.GetHeader().GetStateRootHash(): "
-                    << txBlock.GetHeader().GetStateRootHash());
+    auto res = m_mediator.m_validator->CheckTxBlocks(
+        txBlocks, m_mediator.m_blocklinkchain.GetBuiltDSComm(),
+        m_mediator.m_blocklinkchain.GetLatestBlockLink());
+    switch (res) {
+      case ValidatorBase::TxBlockValidationMsg::VALID:
+        CommitTxBlocks(txBlocks);
+        break;
+      case ValidatorBase::TxBlockValidationMsg::INVALID:
+        LOG_GENERAL(INFO, "[TxBlockVerif]"
+                              << "Invalid blocks");
+        break;
+      case ValidatorBase::TxBlockValidationMsg::STALEDSINFO:
+        LOG_GENERAL(INFO, "[TxBlockVerif]"
+                              << "Saved to buffer");
+        m_txBlockBuffer.clear();
+        for (const auto& txBlock : txBlocks) {
+          m_txBlockBuffer.emplace_back(txBlock);
+        }
+        break;
+      default:;
+    }
+  }
 
-      m_mediator.m_node->AddBlock(txBlock);
+  return true;
+}
 
-      // Store Tx Block to disk
-      if (!ARCHIVAL_NODE) {
-        vector<unsigned char> serializedTxBlock;
-        txBlock.Serialize(serializedTxBlock, 0);
-        BlockStorage::GetBlockStorage().PutTxBlock(
-            txBlock.GetHeader().GetBlockNum(), serializedTxBlock);
-      } else {
-        for (unsigned int i = 0; i < txBlock.GetMicroBlockHashes().size();
-             i++) {
-          if (!txBlock.GetIsMicroBlockEmpty()[i]) {
-            m_mediator.m_archival->AddToFetchMicroBlockInfo(
-                txBlock.GetMicroBlockHashes()[i]);
-          } else {
-            LOG_GENERAL(INFO, "MicroBlock of hash "
-                                  << txBlock.GetMicroBlockHashes()[i].hex()
-                                  << " empty");
-          }
-        }
+void Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
+  LOG_GENERAL(INFO, "[TxBlockVerif]"
+                        << "Success");
+  for (const auto& txBlock : txBlocks) {
+    LOG_EPOCH(
```

Comment:
Can you extract this entire logging message to a new function for code readability

Reply:
OK.